source/rust_verify/example/guide/modes.rs

```diff
@@ -381,6 +381,73 @@ fn add_one(x: u8) -> (ret: u8)
 }
 // ANCHOR_END: const1
 
+// ANCHOR: fib 
+spec fn fib(n: nat) -> nat
+    decreases n,
+{
+    if n == 0 {
+        0
+    } else if n == 1 {
+        1
+    } else {
+        fib((n - 2) as nat) + fib((n - 1) as nat)
+    }
+}
+
+proof fn lemma_fib_is_monotonic(i: nat, j: nat)
+    requires
+        i <= j,
+    ensures
+        fib(i) <= fib(j),
+    decreases j - i,
+{
+    if i < 2 && j < 2 {
+    } else if i == j {
+    } else if i == j - 1 {
+        assert(fib(j) == fib((j - 2) as nat) + fib((j - 1) as nat));
+        lemma_fib_is_monotonic(i, (j - 1) as nat);
+    } else {
+        lemma_fib_is_monotonic(i, (j - 1) as nat);
+        lemma_fib_is_monotonic(i, (j - 2) as nat);
+    }
```

Comment:
There's a simpler proof for this, that we found during the tutorial proof debugging session:

```suggestion
    if i < 2 && j < 2 {
    } else if i == j {
    } else if i == j - 1 {
        reveal_with_fuel(fibo_spec, 2);
    } else {
        reveal_with_fuel(fibo_spec, 2);
        lemma_fib_is_monotonic(i, (j - 1) as nat);
    }
```